src/zjs_ocf_client.c

```diff
@@ -155,18 +159,74 @@ static jerry_value_t get_props_from_response(oc_client_response_t *data)
                                oc_string(rep->name));
             DBG_PRINT("%d\n", rep->value.integer);
             break;
+        case DOUBLE:
+            zjs_obj_add_number(prop_object, (double)rep->value.double_p,
+                               oc_string(rep->name));
+            DBG_PRINT("%lf\n", rep->value.double_p);
+            break;
         case BYTE_STRING:
         case STRING:
             zjs_obj_add_string(prop_object, oc_string(rep->value.string),
                                oc_string(rep->name));
             DBG_PRINT("%s\n", oc_string(rep->value.string));
             break;
+        case INT_ARRAY:
+            DBG_PRINT("[ ");
+            i_array = oc_int_array(rep->value.array);
+            size = oc_int_array_size(rep->value.array);
+            if (size > 0) {
+                ZVAL array = jerry_create_array(size);
```

Comment:
I think you would still want to create/add the array if the `size == 0`. I guess the question is does OCF allow you to send a zero length array. If it does then you should add an empty array to JS.

Reply:
Good question, maybe we need to look at the OCF spec to see if an empty array is allowed.